sdk/python/kfp/dsl/types/type_annotations.py

```diff
@@ -233,13 +233,37 @@ def is_artifact_class(artifact_class_or_instance: Type) -> bool:
 
 
 def is_list_of_artifacts(
-    type_var: Union[Type[List[artifact_types.Artifact]],
-                    Type[artifact_types.Artifact]]
+    annotation: Union[Type[List[artifact_types.Artifact]],
+                      Type[artifact_types.Artifact]]
 ) -> bool:
-    # the type annotation for this function's `type_var` parameter may not actually be a subclass of the KFP SDK's Artifact class for custom artifact types
-    is_list_or_list_generic = getattr(type_var, '__origin__', None) == list
-    # in >= python3.9, List wont have .__args__ if it's used as `-> List` with no inner type argument
-    contains_artifact = hasattr(
-        type_var, '__args__') and type_annotations.is_artifact_class(
-            type_var.__args__[0])
-    return is_list_or_list_generic and contains_artifact
+    """Checks if an object is a list of list of artifacts annotation (e.g.,
+    List[Artifact], List[Dataset])"""
+    return is_generic_list(annotation) and issubclass_of_artifact(
+        get_inner_type(annotation))
+
+
+def get_inner_type(annotation: Any) -> Union[Any, None]:
```

Comment:
nit: `Optional[Any]`?

Reply:
Thank you. Will try to catch these going forward.

Also updated this throughout the codebase.